library/core/src/main/java/com/google/android/exoplayer2/util/Util.java

```diff
@@ -901,7 +901,8 @@ public static int inferContentType(String fileName) {
     } else if (fileName.endsWith(".m3u8")) {
       return C.TYPE_HLS;
     } else if (fileName.endsWith(".ism") || fileName.endsWith(".isml")
-        || fileName.endsWith(".ism/manifest") || fileName.endsWith(".isml/manifest")) {
+        || fileName.endsWith(".ism/manifest") || fileName.endsWith(".isml/manifest")
+            || Uri.parse(fileName).getLastPathSegment().matches("manifest\\(.*\\)")) {
```

Comment:
I don't think we want to match if the .ism or .isml bit isn't there.

Reply:
Would you prefer if this is pushed into a reg-exp as well?

Reply:
I think ideally it would be a regex check directly on fileName (i.e. avoiding parsing as a uri, particularly given the method above converts from a uri into a string to call this method :)).